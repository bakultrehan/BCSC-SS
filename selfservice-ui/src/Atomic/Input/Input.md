This BC Input is amazing, use it responsibly.

## Examples

### Input

Text inputs allow users to enter a single line of text.

```js
<Input label='Input' counter='15' value='test'></Input>
```

<!-- Open console and start typing
click
```vue
import Vue from 'vue';
<template
  ><v-app>
    <Input label="input" @input="input" :value="value"></Input>
  </v-app>
</template>
<script>
export default {
  data() {
    return { value: '' };
  },
  methods: {
    input(val) {
      console.log('you typed value:', val);
    }
  }
};
</script>
``` -->
