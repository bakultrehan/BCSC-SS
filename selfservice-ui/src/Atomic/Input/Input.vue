/** * Input atomic component * Primary buttons allow users to carry out an
important action on your service, such as Download or Submit. */

<template>
  <v-text-field
    :value="value"
    :counter="counter"
    :label="label"
    :required="required"
    @input="input"
    @blur="input"
    v-bind="$attrs"
  ></v-text-field>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

/**
 *
 * @displayName  Input
 */

@Component
export default class Input extends Vue {
  /**
   * The counter for the input.
   *
   */
  @Prop({
    default: ''
  })
  private counter!: string | number;

  /**
   * The type for the input.
   * @values input, password
   */
  @Prop({
    default: 'text'
  })
  private type!: string;

  /**
   *  value for input
   */
  @Prop({
    default: ''
  })
  private value!: string;

  /**
   *  label text
   */
  @Prop({
    default: ''
  })
  private label!: string;

  /**
   *  required
   */
  @Prop({
    default: false
  })
  private required!: boolean;

  /**
   * input
   * @description update value on key up or blur
   * @param {string} inputval
   */
  private input(inputval: string) {
    this.$emit('input', inputval);
  }
}
</script>

<style lang="scss" scoped>
// @import './../../assets/styles/theme.scss';
</style>
