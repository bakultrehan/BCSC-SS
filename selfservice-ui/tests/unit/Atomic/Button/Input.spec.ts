import { mount, createLocalVue } from '@vue/test-utils';
import Input from '@/Atomic/Input/Input.vue';
import Vuetify from 'vuetify';

describe('Input.vue', () => {
  let wrapper: any;

  let vuetify: any;
  const localVue = createLocalVue();

  localVue.use(Vuetify);
  vuetify = new Vuetify();

  const mountFunction = (options: any) => {
    return mount(Input, {
      vuetify,
      localVue,
      ...options
    });
  };

  afterEach(() => {
    wrapper.destroy();
  });

  it('renders props when passed', () => {
    wrapper = mountFunction({
      propsData: {
        input: jest.fn(),
        value: 'test'
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('renders props when passed', async () => {
    wrapper = mountFunction({
      propsData: {
        input: jest.fn(),
        value: 'test'
      }
    });
    const inputElement = wrapper.find('input');
    inputElement.element.value = 'value';
    inputElement.trigger('input');

    await wrapper.vm.$nextTick();

    expect(wrapper.element).toMatchSnapshot();
  });
});
