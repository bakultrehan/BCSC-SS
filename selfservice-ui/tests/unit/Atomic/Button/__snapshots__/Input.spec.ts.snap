// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue renders props when passed 1`] = `
<div
  class="bc-form-text"
  input="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
>
  <label
    class="text-left my-1"
    for=""
  >
    
    
    
    <!---->
  </label>
   
  <!---->
   
  <div
    class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <fieldset
          aria-hidden="true"
        >
          <legend>
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <div
          class="v-text-field__slot"
        >
          <input
            id="input-3"
            input="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Input.vue renders props when passed 2`] = `
<div
  class="bc-form-text"
  input="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
>
  <label
    class="text-left my-1"
    for=""
  >
    
    
    
    <!---->
  </label>
   
  <!---->
   
  <div
    class="v-input input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-text-field--outlined"
  >
    <div
      class="v-input__control"
    >
      <div
        class="v-input__slot"
      >
        <fieldset
          aria-hidden="true"
        >
          <legend>
            <span>
              ​
            </span>
          </legend>
        </fieldset>
        <div
          class="v-text-field__slot"
        >
          <input
            id="input-9"
            input="function mockConstructor() {
        return fn.apply(this, arguments);
      }"
            type="text"
          />
        </div>
      </div>
      <div
        class="v-text-field__details"
      >
        <div
          class="v-messages theme--light"
        >
          <span
            class="v-messages__wrapper"
            name="message-transition"
            tag="div"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
